awx/ui_next/src/screens/Template/JobTemplateEdit/JobTemplateEdit.test.jsx

```diff
@@ -1,36 +1,98 @@
 import React from 'react';
-import { JobTemplatesAPI } from '@api';
-import { mountWithContexts } from '@testUtils/enzymeHelpers';
+import { JobTemplatesAPI, LabelsAPI } from '@api';
+import { mountWithContexts, waitForElement } from '@testUtils/enzymeHelpers';
 import JobTemplateEdit from './JobTemplateEdit';
 
 jest.mock('@api');
 
-describe('<JobTemplateEdit />', () => {
-  const mockData = {
-    id: 1,
-    name: 'Foo',
-    description: 'Bar',
-    job_type: 'run',
-    inventory: 2,
-    project: 3,
-    playbook: 'Baz',
-    type: 'job_template',
-    summary_fields: {
-      user_capabilities: {
-        edit: true,
+const mockJobTemplate = {
+  id: 1,
+  name: 'Foo',
+  description: 'Bar',
+  job_type: 'run',
+  inventory: 2,
+  project: 3,
+  playbook: 'Baz',
+  type: 'job_template',
+  summary_fields: {
+    user_capabilities: {
+      edit: true,
+    },
+    labels: {
+      results: [{ name: 'Sushi', id: 1 }, { name: 'Major', id: 2 }],
+    },
+  },
+};
+
+const mockRelatedCredentials = {
+  count: 2,
```

Comment:
The data from these endpoints can be sprawling and nested - To create these mocks, I took a sampling from the API and removed _most_ fields, only keeping the ones that are relevant for this view.